go.mod

```diff
@@ -269,15 +269,15 @@ require (
 	github.com/Azure/azure-sdk-for-go/sdk/azcore v0.22.0 // indirect
 	github.com/Azure/azure-sdk-for-go/sdk/keyvault/internal v0.2.1 // indirect
 	github.com/AzureAD/microsoft-authentication-library-for-go v0.4.0 // indirect
+	github.com/Microsoft/go-winio v0.5.2 // indirect
 	github.com/chromedp/cdproto v0.0.0-20220208224320-6efb837e6bc2 // indirect
-	github.com/containerd/containerd v1.5.9 // indirect
+	github.com/containerd/containerd v1.6.2 // indirect
```

Comment:
If containerd v1.6.2 is already a required dependency, do we need the replace directive?

Reply:
That was the only way I could remove several vulnerable old versions from `go.sum`. I think this is needed because other versions belongs to other minor releases. Maybe @sakjur can help us here.

Reply:
Having them in go.sum isn’t a problem if they’re not actually placed into the final package (which they won’t be).

Go doesn’t make a difference between minor and patch releases (see https://go.dev/ref/mod#major-version-suffixes ).

You might have some luck pruning the unused versions using the compat flag based on this, but basically inclusion in go.sum doesn’t mean it ends up in the binary, so you don’t have to:

> Because Go 1.16 and earlier did not support module graph pruning, the full transitive closure of dependencies — including transitive go 1.17 dependencies — is still included for each module that specifies go 1.16 or lower. (At go 1.16 and below, the go.mod file includes only [direct dependencies](https://go.dev/ref/mod#glos-direct-dependency), so a much larger graph must be loaded to ensure that all indirect dependencies are included.)
>
> The [go.sum file](https://go.dev/ref/mod#go-sum-files) recorded by [go mod tidy](https://go.dev/ref/mod#go-mod-tidy) for a module by default includes checksums needed by the Go version one below the version specified in its [go directive](https://go.dev/ref/mod#go-mod-file-go). So a go 1.17 module includes checksums needed for the full module graph loaded by Go 1.16, but a go 1.18 module will include only the checksums needed for the pruned module graph loaded by Go 1.17. The -compat flag can be used to override the default version (for example, to prune the go.sum file more aggressively in a go 1.17 module).

Reply:
Great! Thanks for the explanation. I removed the replace directive.